api/errors/builder.go

```diff
@@ -0,0 +1,93 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker/builder component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorCodeAtLeastOneArg is generated when the parser comes across a
+	// Dockerfile command that doesn't have any args.
+	ErrorCodeAtLeastOneArg = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
One of the purposes of this is to have a static string for each error message so that they can, eventually, be translated.  You seem to be pushing for one error code mapping to potentially many error strings. That will break i18n where they need to have an exact 1-1 mapping between error code/ID and a string.  Additionally, with a 1-M mapping, I think you mean to have the exact error string still be hard-coded into the code itself, meaning its (sort of) dynamic and not static. Again, that breaks the ability to programmatically know exactly what error we have.

With i18n an end-user needs to be able to see an error message in some language that they can't read, but take the error ID and look it up in some docs and know exactly what that message says in their chosen language.  They can't do that if its a 1-M mapping.